message_processor.go

```diff
@@ -27,11 +31,40 @@ func (mg messageProcessor) process(lf *logPair) {
 	consumeLog(lf)
 }
 
+type tmpBuffer struct {
+	tBuf                      bytes.Buffer
+	bufferTimer               time.Time
+	bufferReset               bool
+	bufferHoldDurationExpired bool
+}
+
+func appendToTempBuffer(p *tmpBuffer, b *logdriver.LogEntry) {
```

Comment:
this appears it is better refactor it as a member function 
```
func (b *tmpBuffer) append(log *logdriver.LogEntry) error {
...
}
```

Reply:
Done